src/garage/tf/algos/dqn.py

```diff
@@ -236,10 +242,12 @@ def optimize_policy(self, itr, samples_data):
 
         # normalize pixel to range [0, 1] since the samples stored in the
         # replay buffer are of type uint8 and not normalized, for memory
-        # optimization
-        observations = normalize_pixel_batch(self.env_spec, observations)
-        next_observations = normalize_pixel_batch(self.env_spec,
-                                                  next_observations)
+        # optimization. DiscreteCNNQFunction already does this internally.
+
+        if not isinstance(self.qf, DiscreteCNNQFunction):
+            observations = normalize_pixel_batch(observations)
+            next_observations = normalize_pixel_batch(next_observations)
+
         loss, _ = self._train_qf(observations, actions, rewards, dones,
                                  next_observations)
```

Comment:
@ryanjulian does this part look okay? the idea is to avoid normalizing twice and only normalize if the Q function doesn't do so internally (which at the moment is only done by DiscreteCNNQFunction). Alternatively I can make every Q function normalize akro.Images, though I'm not sure that's always desirable. 

If this looks good then it's ready to be merged.

Reply:
this is hacky -- i would prefer if you make it consistent across the codebase. 

an argument can be made either way about whether the QFunction or the algorithm should be normalizing.

without doing a grand review of API design, we should just go with the decision which creates the last amount of change. in this case, that seems that it would be to remove normalization from DiscreteCNNQFunction.

Reply:
note that #734 actually tells you what to do here -- the models (FooCNNBar) should be doing the normalizing, not the algorithms.